app/gui2/src/composables/nodeColors.ts

```diff
@@ -17,14 +17,16 @@ export function useNodeColors(getCssValue: (variable: string) => string) {
     }
   }
 
-  const visibleNodeColors = computed(() => {
+  function getNodeColors(filter?: (node: NodeId) => boolean) {
     const colors = new Set<string>()
     for (const node of graphStore.db.nodeIds()) {
-      const color = getNodeColor(node)
-      if (color) colors.add(color)
+      if (filter?.(node) !== false) {
+        const color = getNodeColor(node)
+        if (color) colors.add(color)
+      }
     }
     return colors
-  })
+  }
```

Comment:
Before, it was a computed value. Perhaps this could return one, so we won't iterate if nothing in filter nor node's color changed?

Reply:
it seems like instead the caller wraps this call in a `computed` now - but yeah, it might make sense to return a `computed` directly if we never want the return value of this to not be reactive